coreax/metrics.py

```diff
@@ -95,86 +91,104 @@ def __init__(self, kernel: coreax.kernel.Kernel, precision_threshold: float = 1e
 
     def compute(
         self,
-        x: ArrayLike,
-        y: ArrayLike,
+        reference_data: _Data,
+        comparison_data: _Data,
         block_size: int | None = None,
-        weights_x: ArrayLike | None = None,
-        weights_y: ArrayLike | None = None,
     ) -> Array:
         r"""
         Calculate maximum mean discrepancy.
 
-        If no weights are given for dataset ``y``, standard MMD is calculated. If
-        weights are given, weighted MMD is calculated. For both cases, if the size of
-        matrix blocks to process is less than the size of both datasets and if
-        ``weights_x`` is given in the weighted case, the calculation
-        is done block-wise to limit memory requirements.
+        If uniform weights are given for dataset ``reference_data``, standard MMD is
+        calculated. If non-uniform weights are given, weighted MMD is calculated. For
+        both cases, if the size of matrix blocks to process is less than the size of
+        both datasets and reference weights are not uniform in the weighted case, the
+        calculation is done block-wise to limit memory requirements.
 
-        :param x: The original :math:`n \times d` data
-        :param y: :math:`m \times d` dataset
+        :param reference_data: An instance of the class :class:`coreax.data.Data`
+        :param comparison_data: An instance of the class :class:`coreax.data.Data` to
+            compare against ``reference_data``
```

Comment:
Similar to above, could point towards the class docs as they have dimensions for the data array and the weights array?

Reply:
Could do, but I do think having the expected dimensions in the function docstring is good for immediate ease-of-use to an end user (especially in an IDE). Also see my response to the similar comment above, but since this is a concrete implementation rather than an abstract base class, we don't have to worry about other `Metric` implementations possibly having different requirements.

Reply:
Agreed, added